pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -138,7 +138,7 @@ func extractStructMarkers(decl *ast.GenDecl, sTyp *ast.StructType, results *mark
 
 	if decl.Doc != nil {
 		for _, comment := range decl.Doc.List {
-			if marker := extractMarker(comment); marker.Value != "" {
+			if marker := extractMarker(comment); marker.Identifier != "" {
```

Comment:
If we are making this based on identifier, I think we need to work out how to handle duplicates.

Looking at the rest of the package, and what you've implemented here, I think we have to do it at the MarkerSet level.

Instead of  `type MarkerSet map[string]Marker` I think we want `type MarkerSet map[string][]Marker`, and then we could handle multiple markers for the same ID, eg checking you don't have multiple `+optional`, or being able to tell the difference between multiple `XValidation` rules

Otherwise this seems sensible and is similar to how controller-tools is doing this